packages/stargate/src/signingstargateclient.ts

```diff
@@ -198,6 +206,44 @@ export class SigningStargateClient extends StargateClient {
     return this.signAndBroadcast(senderAddress, [sendMsg], this.fees.send, memo);
   }
 
+  public async delegateTokens(
+    senderAddress: string,
+    validatorAddress: string,
+    amount: Coin,
+    memo = "",
+  ): Promise<BroadcastTxResponse> {
+    const delegateMsg = {
+      typeUrl: "/cosmos.staking.v1beta1.MsgDelegate",
+      value: MsgDelegate.fromPartial({ delegatorAddress: senderAddress, validatorAddress, amount }),
+    };
+    return this.signAndBroadcast(senderAddress, [delegateMsg], this.fees.delegate, memo);
+  }
+
+  public async undelegateTokens(
+    senderAddress: string,
+    validatorAddress: string,
+    amount: Coin,
+    memo = "",
+  ): Promise<BroadcastTxResponse> {
+    const undelegateMsg = {
+      typeUrl: "/cosmos.staking.v1beta1.MsgUndelegate",
+      value: MsgUndelegate.fromPartial({ delegatorAddress: senderAddress, validatorAddress, amount }),
+    };
+    return this.signAndBroadcast(senderAddress, [undelegateMsg], this.fees.undelegate, memo);
+  }
+
+  public async withdrawRewards(
+    senderAddress: string,
```

Comment:
I'd call this `delegatorAddress`, which is signer but not sender as in token transfer source.